sdks/java/io/jms/src/main/java/org/apache/beam/sdk/io/jms/JmsIO.java

```diff
@@ -667,11 +667,8 @@ public boolean advance() throws IOException {
       try {
         Message message;
         synchronized (this) {
-          if (receiveTimeoutMillis == 0L) {
-            message = this.consumer.receiveNoWait();
-          } else {
-            message = this.consumer.receive(receiveTimeoutMillis);
-          }
+          long timeout = (receiveTimeoutMillis == 0L) ? 1L : receiveTimeoutMillis;
+          message = this.consumer.receive(timeout);
```

Comment:
Could you explain why we need this change? It seems like we are no longer respecting the set receive timeout

Reply:
Yeah let's revert JmsIO change. This is a long standing flaky test likely due to test infra (in-memory jms server)